internal/services/oracle/oracle_exascale_db_storage_vault_resource.go

```diff
@@ -0,0 +1,331 @@
+				return err
+			}
+
+			var model ExascaleDbStorageVaultResourceModel
+			if err = metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding err: %+v", err)
+			}
+
+			_, err = client.Get(ctx, *id)
+			if err != nil {
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			update := &exascaledbstoragevaults.ExascaleDbStorageVaultTagsUpdate{}
+
+			if metadata.ResourceData.HasChange("tags") {
+				update.Tags = pointer.To(model.Tags)
+			}
+
+			err = client.UpdateThenPoll(ctx, *id, *update)
+			if err != nil {
+				return fmt.Errorf("updating %s: %v", id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (ExascaleDbStorageVaultResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			id, err := exascaledbstoragevaults.ParseExascaleDbStorageVaultID(metadata.ResourceData.Id())
+			if err != nil {
+				return fmt.Errorf("retrieving %s: %+v", id, err)
+			}
+
+			client := metadata.Client.Oracle.OracleClient.ExascaleDbStorageVaults
+			result, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(result.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("retrieving %s: %+v", id, err)
+			}
+
+			state := ExascaleDbStorageVaultResourceModel{
+				Name:              id.ExascaleDbStorageVaultName,
+				ResourceGroupName: id.ResourceGroupName,
+			}
+
+			if model := result.Model; model != nil {
+				state.Location = location.Normalize(model.Location)
+				state.Tags = pointer.From(model.Tags)
+				state.Zones = pointer.From(model.Zones)
+				if props := model.Properties; props != nil {
+					state.Name = pointer.ToString(result.Model.Name)
+					state.Location = result.Model.Location
+					state.Tags = pointer.From(model.Tags)
+					state.Zones = pointer.From(result.Model.Zones)
+					state.DisplayName = props.DisplayName
+					state.AdditionalFlashCacheInPercent = pointer.From(props.AdditionalFlashCacheInPercent)
+					state.Description = pointer.From(props.Description)
+					state.HighCapacityDatabaseStorage = ExpandHighCapacityDatabaseStorageInterface(metadata.ResourceData.Get("high_capacity_database_storage").([]interface{}))
+					state.HighCapacityDatabaseStorageInput = ExpandHighCapacityDatabaseStorageInputInterface(metadata.ResourceData.Get("high_capacity_database_storage_input").([]interface{}))
+					state.TimeZone = pointer.From(props.TimeZone)
+				}
+			}
+			return metadata.Encode(&state)
+		},
+	}
+}
+
+func (ExascaleDbStorageVaultResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Oracle.OracleClient.ExascaleDbStorageVaults
+
+			id, err := exascaledbstoragevaults.ParseExascaleDbStorageVaultID(metadata.ResourceData.Id())
+			if err != nil {
+				return fmt.Errorf("retrieving %s: %+v", id, err)
```

Comment:
This is not retrieving the resource, but parsing the resource id:

```suggestion
				return err
```

Reply:
Updated